rfcs/rework-index-updates.md

```diff
@@ -0,0 +1,60 @@
+# Summary
+
+This RFC proposes moving away from the current model of fetching new releases
+to build, moving from running [crates-index-diff] in a thread to using webhooks
+and the [crates-index] crate.
+
+# Motivation
+
+While the current approach has worked well for us so far, it has some problems:
+
+* Running the update in a cronjob every 2 minutes is wasteful, as there is
+  often a greater delay between two publishes.
+* Running the update in a crobjob every 2 minutes adds delay to getting the
+  documentation built if the queue is empty, as the release might potentially
+  have to wait those extra two minutes.
+* The approach doesn't scale, if we want to move to a setup where there is more
+  than a single frontend server we'd have to elect which server runs the fetch.
```

Comment:
Not necessarily, it seems like it'd be relatively easy to make it so that two services racing to add a crate to the queue only add it once (in fact, that's mentioned as being part of the implementation of the proposed method as well).

Reply:
Hmm, that's true, even though races would happen every time with the current solution instead of just when two crates are published at the same time with this RFC.

Reply:
If we go for the hybrid "use webhooks to trigger the diff" approach then I feel like we get best of both worlds, minimal updates being performed, and minimal work to do each update.